sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/sas/TableAccountSasPermission.java

```diff
@@ -0,0 +1,409 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables.sas;
+
+import com.azure.data.tables.implementation.StorageConstants;
+
+import java.util.Locale;
+
+/**
+ * This is a helper class to construct a string representing the permissions granted by an Account SAS. Setting a value
+ * to true means that any SAS which uses these permissions will grant permissions for that operation. Once all the
+ * values are set, this should be serialized with {@code toString()} and set as the permissions field on an
+ * {@link TableAccountSasSignatureValues} object.
+ *
+ * <p>
+ * It is possible to construct the permissions string without this class, but the order of the permissions is particular
+ * and this class guarantees correctness.
+ * </p>
+ *
+ * @see TableAccountSasSignatureValues
+ * @see <a href="https://docs.microsoft.com/rest/api/storageservices/create-account-sas">Create account SAS</a>
+ */
+public final class TableAccountSasPermission {
```

Comment:
I think this type should be removed in favor of `AccountSasPermission` from `azure-storage-common`.

Reply:
Please see my comment above.